README.md

```diff
@@ -320,6 +321,42 @@ GoodJob.retry_on_unhandled_error = false
 GoodJob.on_thread_error = -> (exception) { Raven.capture_exception(exception) }
 ```
 
+### Production setup
+
+Running GoodJob successfully in production environments requires understanding
+a couple of things:
+ * the [execution mode](execute-jobs-async--in-process)
+ * the [database connection pool size](#database-connections)
+ * the [health check probes](#cli-http-health-check-probes) and potentially the
+   [instrumentation support](#monitor-and-preserve-worked-jobs)
+
+The number of the database connections your setup needs is directly dependent
+on the execution mode. A good way to estimate the size of the connections pool
+per process running as `:external` execution mode is:
+
+ * 1 connection dedicated to the scheduler aka `LISTEN/NOTIFY`
+ * 1 connection per thread * `GOOD_JOB_MAX_THREADS`
+ * 1 connection per every extra thread if your application runs these as part
+   of a job
+ * 1 connection dedicated to the cron scheduler if you're running it
+ * 1 connection dedicated to the cron executor if you're running cron
+
+If you're running the queue along with the Rails process, please adjust
+the pool size taking into account the number of web server threads!
+
+It is important to understand that the queue process won't crash if the
+connections pool is exhausted, instead it will report an exception (eg.
+`ActiveRecord::ConnectionTimeoutError`)!
```

Comment:
I'd like this to be isolated inside of the section on database connection pool size. 

Overall, I'm imagining the section you're proposing here is more of a checklist of items that explains _why_ you should configure them for production, but the _how and details_ lives in the section related to that specific feature.

Reply:
@bensheldon thanks for reviewing this, I followed your suggestions and moved it as the _Database Connections_ sub-section.

As for the reasoning behind it, we basically were wrapping our heads around trying to compile a production-ready setup and still failed to grasp the whole connections usage part. It can be seen as a _checklist_, though to me, I just want to make sure our findings are documented, as I'll definitely be returning to them as I use `good_jobs` in more projects :wink: